### PR TITLE
Questionnaire

### PR DESCRIPTION
**Pull request by mnsison**
Hi, could you please review my code? Thank you.

### CODE REVIEW

File: src/Questionnaire.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    this.handleSubmit =this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event){
+  console.log('sad')
+    event.preventDefault();
+    if(this.state.response.length === 16){ //checks if all questions have been answered
+      const id =45;//test ID
+      console.log(id);
+      this.storeAnswer(id);
```
Comment (chalynpai) [inline src/Questionnaire.js]: would be good to return a warning which questions have been unanswered?

**Review by chalynpai — COMMENTED** [review]
Suggestion on questionnaire feedback

File: README.md [hunk 2]
```diff
@@ -1,4 +1,33 @@
...
+
+`pip install mysql.connector `
+
+`pip install flask_cors`
+
+`pip install flask`
+
+How to run locally:
+
+Front-end
```
Comment (wclui) [inline README.md]: should've pulled and merge in development so these will not be new changes

File: server/questionnaire.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+            cursor = connection.cursor(buffered=True)
+
+            sql = '''INSERT INTO Questionnaire(userId, response1, response2, response3, response4, response5, response6, response7, response8, response9, response10, response11, response12, response13, response14, response15, response16)
+                VALUES (%s, %s, %s, %s, %s, %s, %s, %s,%s, %s, %s, %s, %s, %s, %s, %s, %s);'''
+            values = (response[17], response[0], response[1], response[2], response[3], response[4], response[5], response[6], response[7], response[8],
+                    response[9], response[10], response[11], response[12], response[13], response[14], response[15], response[16])
+            cursor.execute(sql, values)
+            connection.commit()
+            cursor.close()
+            return{"response":"Success"}
```
Comment (wclui) [inline server/questionnaire.py]: space after return

File: src/Questionnaire.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        headers: {'Content-Type':'application/json'},
+        body: JSON.stringify({"responses":this.state.response}),
+      });
+
+      fetch(myRequest)
+      .then((res) =>
+        res.json())
+      .then((res) =>{
+        if(res.response === "Success")
+          alert("Thank you for answering the questionnaire!");
```
Comment (wclui) [inline src/Questionnaire.js]: should this lead somewhere? maybe lead to the swiping or profile page?

File: src/Questionnaire.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      response: this.state.response.concat(answer)});
+  }
+  render(){
+    return(
+      <div>
+        <div>
+          <h1>PP Friends Questionnaire</h1>
+          <h4>Please answer all questions before submitting</h4>
+        </div>
+        {this.state.questionBank.length > 0 && this.state.response.length < 16 &&
```
Comment (wclui) [inline src/Questionnaire.js]: what's the purpose of checking this.state.response.length here?

**Review by wclui — CHANGES_REQUESTED** [review]

File: src/Questionnaire.js [hunk 6]
```diff
@@ -15,25 +15,27 @@ class Questionnaire extends Component {
...
 
   handleSubmit(event){
-  console.log('sad')
     event.preventDefault();
-    if(this.state.response.length === 16){ //checks if all questions have been answered
-      const id =45;//test ID
-      console.log(id);
-      this.storeAnswer(id);
-
+    if(this.state.response.length >= 16){ //checks if all questions have been answered
```
Comment (wclui) [inline src/Questionnaire.js:19]: what's the > here for

**Review by wclui — APPROVED** [review]

**Pull request merged**